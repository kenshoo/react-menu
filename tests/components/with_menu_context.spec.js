import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import WithMenuContext from "../../src/components/with_menu_context";
import Menu from "../../src/components/menu";

const mockCmp = () => <div>TEST</div>;
const MockCmpWithMenuContext = WithMenuContext(mockCmp);

describe("WithMenuContext HOC", () => {
  test("renders wrapped component", () => {
    const cmp = renderer.create(
      <Menu>
        <MockCmpWithMenuContext />
      </Menu>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("adds context to children", () => {
    const cmp = mount(
      <Menu>
        <MockCmpWithMenuContext />
      </Menu>
    );

    expect(cmp.find(mockCmp).props().onSelectElement).toBeTruthy();
  });
});
