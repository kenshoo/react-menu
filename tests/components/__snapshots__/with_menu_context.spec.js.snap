// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithMenuContext HOC renders wrapped component 1`] = `
<div
  className="menu"
>
  <div>
    TEST
  </div>
</div>
`;
